src/libraries/System.Private.CoreLib/src/System/SpanHelpers.cs

```diff
@@ -404,5 +407,230 @@ public static unsafe void ClearWithReferences(ref IntPtr ip, nuint pointerSizeLe
+                    //     +---------------+
+                    //     | A | B | C | D |
+                    //     +---------------+
+                    //         Shuffle --->
+                    //     +---------------+
+                    //     | B | A | D | C |
+                    //     +---------------+
+                    //         Permute --->
+                    //     +---------------+
+                    //     | D | C | B | A |
+                    //     +---------------+
+                    tempFirst = Avx2.Shuffle(tempFirst, ReverseMask);
+                    tempFirst = Avx2.Permute2x128(tempFirst, tempFirst, 1);
+                    tempLast = Avx2.Shuffle(tempLast, ReverseMask);
+                    tempLast = Avx2.Permute2x128(tempLast, tempLast, 1);
+
+                    // Store the reversed vectors
+                    Unsafe.As<byte, Vector256<byte>>(ref first) = tempLast;
+                    Unsafe.As<byte, Vector256<byte>>(ref last) = tempFirst;
+
+                    // Adjust the references to point to the next vector
+                    first = ref Unsafe.Add(ref first, Vector256<byte>.Count);
+                    last = ref Unsafe.Add(ref last, -Vector256<byte>.Count);
+                    numBytesWritten += Vector256<byte>.Count * 2;
+                } while (numBytes - numBytesWritten >= Vector256<byte>.Count * 2);
+            }
+            else if (Ssse3.IsSupported && Vector128<byte>.Count * 2 <= numBytes)
+            {
+                Vector128<byte> ReverseMask = Vector128.Create((byte)8, 9, 10, 11, 12, 13, 14, 15, 0, 1, 2, 3, 4, 5, 6, 7);
+                last = ref Unsafe.Add(ref Unsafe.Add(ref first, numBytes), -Vector128<byte>.Count);
+                do
+                {
+                    // Load in values from beginning and end of the array.
+                    Vector128<byte> tempFirst = Unsafe.As<byte, Vector128<byte>>(ref first);
+                    Vector128<byte> tempLast = Unsafe.As<byte, Vector128<byte>>(ref last);
+
+                    // Shuffle to reverse each vector:
+                    //     +-------+
+                    //     | A | B |
+                    //     +-------+
+                    //          --->
+                    //     +-------+
+                    //     | B | A |
+                    //     +-------+
+                    tempFirst = Ssse3.Shuffle(tempFirst, ReverseMask);
+                    tempLast = Ssse3.Shuffle(tempLast, ReverseMask);
+
+                    // Store the reversed vectors
+                    Unsafe.As<byte, Vector128<byte>>(ref first) = tempLast;
+                    Unsafe.As<byte, Vector128<byte>>(ref last) = tempFirst;
+
+                    // Adjust the references to point to the next vector
+                    first = ref Unsafe.Add(ref first, Vector128<byte>.Count);
+                    last = ref Unsafe.Add(ref last, -Vector128<byte>.Count);
```

Comment:
Semantically, using [Subtract](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.unsafe.subtract?view=net-6.0) would be better.
```suggestion
                    last = ref Unsafe.Subtract(ref last, Vector128<byte>.Count);
```

Reply:
Updated.